src/main/java/Duke.java

```diff
@@ -1,10 +1,150 @@
-public class Duke {
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import javafx.application.Application;
+import javafx.scene.Scene;
+import javafx.scene.control.Button;
+import javafx.scene.control.Label;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.control.TextField;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.Region;
+import javafx.scene.layout.VBox;
+import javafx.stage.Stage;
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
+
+public class Duke extends Application {
+
+    private ScrollPane scrollPane;
+    private VBox dialogContainer;
+    private TextField userInput;
+    private Button sendButton;
+    private Scene scene;
+    private Image user = new Image(this.getClass().getResourceAsStream("/images/User.png"));
+    private Image duke = new Image(this.getClass().getResourceAsStream("/images/Duke.png"));
+
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+    private String filePath = "data/tasks.txt";
+
+    public Duke() {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (FileNotFoundException e) {
+            storage.initialiseData();
+        } catch (IOException e) {
+            ui.showLoadingError(e.getMessage());
+            tasks = new TaskList();
+        }
+    }
+
+    @Override
+    public void start(Stage stage) {
+        // Step 1. Setting up required components
```

Comment:
Is this comment really necessary? Would you be able to tell what you are doing here even without it?